pkgs/development/tools/regclient/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+let bins = [ "regbot" "regctl" "regsync" ]; in
+
+buildGoModule rec {
+  pname = "regclient";
+  version = "0.3.8";
+  tag = "v${version}";
+  commit = "c0d4e8078e3e40d9854010a52e8353f98c8ae1ed";
+
+  src = fetchFromGitHub {
+    owner = "regclient";
+    repo = "regclient";
+    rev = tag;
+    sha256 = "14w0g24sgphgib33sdvrvwk86p7km2pasb5fmr3p48i7sc71ja3h";
+  };
+  vendorSha256 = "sha256-9sRjP7lxMRdt9D9ElIX+mbYIvCaknWMgDyYl+1/q0/g=";
+
+  outputs = [ "out" ] ++ bins;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/regclient/regclient/regclient.VCSRef=${commit}"
+    "-X main.VCSRef=${commit}"
+    "-X main.VCSTag=${tag}"
+  ];
+
+  postInstall = lib.concatStringsSep "\n" (
+      builtins.map (bin: "mkdir -p \$${bin}/bin && mv $out/bin/${bin} \$${bin}/bin/ && ln -s \$${bin}/bin/${bin} $out/bin/") bins
+    );
```

Comment:
Using a for loop here would make this a lot clearer and easier to understand.

Reply:
Tried bash but reverted. Refactored instead.